django/cantusdb_project/main_app/views/chant.py

```diff
@@ -813,19 +829,36 @@ def get_queryset(self) -> QuerySet:
             # Filter the QuerySet with Q object
             chant_set = chant_set.filter(q_obj_filter)
             sequence_set = sequence_set.filter(q_obj_filter)
+            # Fetch only the values necessary for rendering the template
+            chant_set = chant_set.values(*values_for_rendering_template)
+            sequence_set = sequence_set.values(*values_for_rendering_template)
             # Finally, do keyword searching over the querySet
             if self.request.GET.get("keyword"):
                 keyword = self.request.GET.get("keyword")
-                # the operation parameter can be "contains" or "starts_with"
-                if self.request.GET.get("op") == "contains":
-                    chant_set = keyword_search(chant_set, keyword)
-                    sequence_set = keyword_search(sequence_set, keyword)
+                operation = self.request.GET.get("op")
+                if operation == "contains":
```

Comment:
This is a case where type annotations would help understand a bit better... 

```
operation: Optional[str] = self.request.GET.get("op")
if operation and operation == "contains":
    ....
```

IOW, you should check the `None` case as well, since if `op` isn't in your request parameters, `GET.get("op")` will return `None`.

Reply:
I like this suggestion - let me make this change